### PR TITLE
fix: usage of `a deprecated Node.js version`

### LINKED ISSUE #2380: `.github/workflows/postrelease.yml` uses `a deprecated Node.js version`
[`.github/workflows/postrelease.yml` uses `a deprecated Node.js version`](https://github.com/restatedev/restate/actions/runs/11919971674).

### PR DESCRIPTION
**Pull request by hirehamir**
Fixes https://github.com/restatedev/restate/issues/2380

### CODE REVIEW

File: .github/workflows/update.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
 
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
```
Comment (hirehamir) [inline .github/workflows/update.yml:16]: This is the latest major version at the time of this writing.

**Comment by hirehamir** [comment]
@tillrohrmann are you open to this change?

**Review by tillrohrmann — APPROVED** [review]
Thanks a lot for updating the actions/checkout action @hamirmahal and sorry for missing it. The changes look good to me. Merging it now :-)

**Comment by hirehamir** [comment]
You're welcome! I'm happy to help.

**Pull request merged**